### PR TITLE
Refactor(eos_designs): Use VRF ID instead of VRF VNI as offset for evpn underlay l3 multicast group

### PR DESCRIPTION
**Pull request by Vibhu-gslab**
## Change Summary
Change algorithm to use VRF ID instead of VRF VNI as offset. (in evpn_underlay_l3_multicast_group()) 

## Related Issue(s)

Fixes #https://github.com/aristanetworks/avd-internal/issues/161

## Component(s) name

`arista.avd.eos_designs`

## Proposed changes
replaced vrf_vni with vrf_id in evpn_underlay_l3_multicast_group()

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [ ] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/docs/release-notes/5.x.x.md [hunk 1]
```diff
@@ -176,6 +176,12 @@ Starting AVD 5.0.0, by default `redistribute igmp` will only get configured when
 
 See the [Porting guide for AVD 5.x.x](../porting-guides/5.x.x.md#removed-redistribute-igmp-from-bgp-vlan-config-belong-to-a-vrf-with-evpn-multicast)
 
+#### vrf_id will be used instead of vrf_vni to get the ip address of evpn underlay multicast group
+
+Starting AVD 5.0.0, `vrf_vni` will be replaced with `vrf_id` in the algorithm to to set the offset for ip address of evpn underlay multicast group when evpn_l3_multicast is enabled.
```
Comment (Shivani-gslab) [inline ansible_collections/arista/avd/docs/release-notes/5.x.x.md]:
```suggestion
Starting AVD 5.0.0, `vrf_id` is used instead of `vrf_vni` in the algorithm to set the offset for IP address of EVPN underlay multicast group when `evpn_l3_multicast` is enabled.
```

File: ansible_collections/arista/avd/docs/porting-guides/5.x.x.md [hunk 2]
```diff
@@ -108,6 +108,55 @@ tenants:
 +             always_redistribute_igmp: true
 ```
 
+### `vrf_id` will be used instead of `vrf_vni` to get the ip address of evpn underlay multicast group
+
+AVD versions below 5.0.0 uses `vrf_vni` in the algorithm to set the offset in order to get the ip address of evpn underlay multicast group.
+
+In AVD version 5.0.0 `vrf_vni` will be replaced with `vrf_id` to set the offset for ip address of evpn underlay multicast group when evpn_l3_multicast is enabled.
+
```
Comment (Shivani-gslab) [inline ansible_collections/arista/avd/docs/porting-guides/5.x.x.md]:
```suggestion
AVD versions below 5.0.0 uses `vrf_vni` in the algorithm to set the offset in order to get the IP address of EVPN underlay multicast group.

In AVD version 5.0.0, `vrf_id` is used instead of `vrf_vni` to set the offset for IP address of EVPN underlay multicast group when `evpn_l3_multicast` is enabled.

```

**Comment by laxmikantchintakindi** [comment]
Needed to update the description of key `evpn_l3_multicast` in `defs_network_services_schema`.

**Review by Shivani-gslab — DISMISSED** [review]
LGTM

**Review by laxmikantchintakindi — DISMISSED** [review]

**Comment by laxmikantchintakindi** [comment]
LGTM

File: ansible_collections/arista/avd/docs/porting-guides/5.x.x.md [hunk 3]
```diff
@@ -108,6 +108,55 @@ tenants:
 +             always_redistribute_igmp: true
 ```
 
+### `vrf_id` will be used instead of `vrf_vni` to get the ip address of evpn underlay multicast group
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/docs/porting-guides/5.x.x.md]:
```suggestion
### `vrf_id` is used instead of `vrf_vni` to calculate the IP address of EVPN underlay multicast group for a VRF
```

File: ansible_collections/arista/avd/docs/porting-guides/5.x.x.md [hunk 4]
```diff
@@ -108,6 +108,55 @@ tenants:
...
+
+AVD versions below 5.0.0 uses `vrf_vni` in the algorithm to set the offset in order to get the IP address of EVPN underlay multicast group.
+
+In AVD version 5.0.0, `vrf_id` is used instead of `vrf_vni` to set the offset for IP address of EVPN underlay multicast group when `evpn_l3_multicast` is enabled.
+
+Inputs:
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/docs/porting-guides/5.x.x.md]: The configurations will only change in case `vrf_id` and `vrf_vni` are set to different values.

This should show how to revert the changes if needed.

I would use structured_config under the VRF (show example).

File: python-avd/pyavd/_eos_designs/structured_config/network_services/vxlan_interface.py [hunk 5]
```diff
@@ -188,7 +188,6 @@ def _get_vxlan_interface_config_for_vrf(self: AvdStructuredConfigNetworkServices
                     underlay_l3_mcast_group_ipv4_pool_offset = get(tenant, "evpn_l3_multicast.evpn_underlay_l3_multicast_group_ipv4_pool_offset", default=0)
                     vrf_data["multicast_group"] = self.shared_utils.ip_addressing.evpn_underlay_l3_multicast_group(
                         underlay_l3_multicast_group_ipv4_pool,
-                        vni,
```
Comment (ClausHolbechArista) [inline python-avd/pyavd/_eos_designs/structured_config/network_services/vxlan_interface.py:191]: Please do not change the signature of the function. These ip_addressing methods can be overridden by expert users with custom Python modules.

File: python-avd/pyavd/api/ip_addressing/__init__.py [hunk 6]
```diff
@@ -314,16 +314,11 @@ def vrf_loopback_ip(self, pool: str) -> str:
     def evpn_underlay_l3_multicast_group(
         self,
         underlay_l3_multicast_group_ipv4_pool: str,
-        vrf_vni: int,
-        vrf_id: int,  # pylint: disable=unused-argument # noqa: ARG002
+        vrf_id: int,
```
Comment (ClausHolbechArista) [inline python-avd/pyavd/api/ip_addressing/__init__.py:318]: Please do not change the signature of the function. These ip_addressing methods can be overridden by expert users with custom Python modules. Just move the pylint/noqa to the vrf_vni field.

**Review by ClausHolbechArista — CHANGES_REQUESTED** [review]

File: python-avd/pyavd/api/ip_addressing/__init__.py (hunk 6 above)
Comment (Vibhu-gslab) [reply python-avd/pyavd/api/ip_addressing/__init__.py:318]: Fixed!

File: python-avd/pyavd/_eos_designs/structured_config/network_services/vxlan_interface.py (hunk 5 above)
Comment (Vibhu-gslab) [reply python-avd/pyavd/_eos_designs/structured_config/network_services/vxlan_interface.py:191]: Fixed

File: ansible_collections/arista/avd/docs/porting-guides/5.x.x.md (hunk 4 above)
Comment (Vibhu-gslab) [reply ansible_collections/arista/avd/docs/porting-guides/5.x.x.md]: Fixed!

File: python-avd/pyavd/api/ip_addressing/__init__.py (hunk 6 above)
Comment (Vibhu-gslab) [reply python-avd/pyavd/api/ip_addressing/__init__.py:318]: sonarcloud is failing after including the variable, can we move it to Accepted issues?

Comment (ClausHolbechArista) [reply python-avd/pyavd/api/ip_addressing/__init__.py:318]: I have accepted the issue.

**Review by ClausHolbechArista — APPROVED** [review]

File: ansible_collections/arista/avd/docs/porting-guides/5.x.x.md [hunk 7]
```diff
@@ -108,6 +108,26 @@ tenants:
...
+tenants:
+  - name: Tenant_C
+    evpn_l3_multicast:
+      enabled: true
+    vrfs:
+      - name: TEN_C_L3_MULTICAST_ENABLED_130_131
+        vrf_vni: 31
++       vrf_id: 31
+        <...>
+```
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/docs/porting-guides/5.x.x.md]: This is not a great suggestion, since it will affect many other things. We should either suggest a structured config snip, or provide a knob to set the group directly under the VRF.

**Review by ClausHolbechArista — CHANGES_REQUESTED** [review]

File: ansible_collections/arista/avd/docs/porting-guides/5.x.x.md (hunk 7 above)
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/docs/porting-guides/5.x.x.md]: The knob should be `evpn_l3_multicast.evpn_underlay_l3_multicast_group`.

Comment (Vibhu-gslab) [reply ansible_collections/arista/avd/docs/porting-guides/5.x.x.md]: Added knob.

**Review by ClausHolbechArista — APPROVED** [review]

**Review by MaheshGSLAB — APPROVED** [review]
